webpages/signup.php

```diff
@@ -1,157 +1,211 @@
-<!DOCTYPE html>
-<html lang="en" >
-<head>
-  <meta charset="UTF-8">
-  <title>Signup</title>
-  
-  
-
-  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css">
-
-  
-      <link rel="stylesheet" href="../css/signup.css">
-
-  
-</head>
-
-<body>
 <?php
-$email = $password = $password2 = $fname = $lname = $gender  ="";
-$emailerr = $passerr = $pass2err = $fnameerr = $lnameerr  ="";
-$flag1 = $flag2 = $flag3 = $flag4 =$flag5 = false;  
-function prepare($data){
-
-  $data = trim($data);
-  $data = stripslashes($data);
-   $data = htmlspecialchars($data);
-  return $data;}
-
-  if ($_SERVER["REQUEST_METHOD"] == "POST")
-  {
-    if(empty($_POST["email"]))
-    {
-      $emailerr = "<div class=\"error\">  * Email is required! </div>";
-    }
-  else{
-
-    $email = prepare($_POST["email"]);
-        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
-  $emailerr = "<div class=\"error\"> Invalid email format </div>";
-}
-else {
-  $emailerr="";
-  $flag1 = true;
+session_start();
+if (!empty($_SESSION["uid"]))
+{
+    header('location: /webpages/home.php');
 }
+?>
+<!DOCTYPE html>
+<html lang="en" >
+    <head>
+        <meta charset="UTF-8">
+        <title>Signup</title>
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css">
+        <link rel="stylesheet" href="../css/signup.css">
+    </head>
+    <body>
+        <?php
+            $email = $password = $password2 = $fname = $lname = $gender  ="";
+            $emailerr = $passerr = $pass2err = $fnameerr = $lnameerr  ="";
+            $flag1 = $flag2 = $flag3 = $flag4 =$flag5 = 0;  
+            function prepare($data){
+            
+              $data = trim($data);
+              $data = stripslashes($data);
+               $data = htmlspecialchars($data);
+              return $data;
+            }
+            
+              if ($_SERVER["REQUEST_METHOD"] === "POST")
+              {
+                if(empty($_POST["email"]))
+                {
+                  $emailerr = "<div class=\"error\">  * Email is required! </div>";
+                }
+              else{
+            
+                $email = prepare($_POST["email"]);
+                    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+              $emailerr = "<div class=\"error\"> Invalid email format </div>";
+            }
+            else {
+              $emailerr="";
+              $flag1 = 1;
+            }
+            
+                }
+                if(empty($_POST["pass"]))
+                {
+                  $passerr = "<div class=\"error\">  * Password is required! </div>";
+                }
+              else{
+                $password = prepare($_POST["pass"]);
+                if (strlen((string)$password)<6)
```

Comment:
do not typecast $password to string rather check first if it is a string and then check if the length is <6